applications/FluidDynamicsApplication/custom_strategies/strategies/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h

```diff
@@ -813,83 +795,74 @@ namespace Kratos {
          */
         void PeriodicConditionProjectionCorrection(ModelPart& rModelPart)
         {
-            if (mrPeriodicIdVar.Key() != 0)
-            {
-                for (typename ModelPart::ConditionIterator itCond = rModelPart.ConditionsBegin(); itCond != rModelPart.ConditionsEnd(); itCond++ )
-                {
-                    ModelPart::ConditionType::GeometryType& rGeom = itCond->GetGeometry();
-                    if (rGeom.PointsNumber() == 2)
+            const int num_nodes = rModelPart.NumberOfNodes();
+            const int num_conditions = rModelPart.NumberOfConditions();
+
+            #pragma omp parallel for
+            for (int i = 0; i < num_nodes; i++) {
+                auto it_node = rModelPart.NodesBegin() + i;
+
+                it_node->GetValue(NODAL_AREA) = 0.0;
+                it_node->GetValue(ADVPROJ) = array_1d<double,3>(3,0.0);
```

Comment:
I would use ZeroVector(3).

the arguement is that the array(3,0.0) constructor may not be supported by AMatrix (correct @pooyan-dadvand ?)

Reply:
the variable is an `array_1d<double,3>` variable. What does this have to do with AMatrix?